src/tilesource.js

```diff
@@ -709,6 +709,166 @@ $.TileSource.prototype = {
                y >= 0 &&
                x < numTiles.x &&
                y < numTiles.y;
+    },
+
+    /**
+     * Decide whether tiles have transparency: this is crucial for
+     * @return {boolean} true if the image has transparency
+     */
+    hasTransparency: function(context2D, url, ajaxHeaders, post) {
+        return !!context2D || url.match('.png');
+    },
+
+    /**
+     * Download tile data
+     * @param {object} context job context that you have to call finish(...) on. It also contains abort(...) function
+     *   that can be called to abort the job.
+     * @param {String} [context.src] - URL of image to download.
+     * @param {String} [context.loadWithAjax] - Whether to load this image with AJAX.
+     * @param {String} [context.ajaxHeaders] - Headers to add to the image request if using AJAX.
+     * @param {String} [context.crossOriginPolicy] - CORS policy to use for downloads
+     * @param {String} [context.postData] - HTTP POST data (usually but not necessarily in k=v&k2=v2... form,
+     *      see TileSrouce::getPostData) or null
+     * @param {Function} [context.callback] - Called once image has been downloaded.
+     * @param {Function} [context.abort] - Called when this image job is aborted.
+     * @param {Number} [context.timeout] - The max number of milliseconds that this image job may take to complete.
+     */
+    downloadTileStart: function (context) {
+        context.image = new Image();
```

Comment:
It's worth calling out in the documentation the fact that it's okay to add your own properties to the `context` object for use in the other methods it gets sent to. 

Actually, what might be nice for context to have a `userData` object that you can attach things to, rather than directly adding properties to `context`. That way you aren't risking clobbering something important with a name collision.